packages/shopify/src/api/endpoints/wishlist/add-item.ts

```diff
@@ -0,0 +1,84 @@
+import type { WishlistEndpoint } from '.'
+import { customerUpdateMutation, getCustomerId } from '../../../utils'
+import type {
+  MutationCustomerUpdateArgs,
+  Mutation,
+  MetafieldInput,
+} from '../../../../admin-schema'
+import { WishlistItem } from './../../../types/wishlist'
+
+const addWishlistItem: WishlistEndpoint['handlers']['addItem'] = async ({
+  res,
+  body: { variables: item, customerToken },
+  config,
+  commerce,
+}) => {
+  if (!item) {
+    return res.status(400).json({
+      data: null,
+      errors: [{ message: 'Missing item!' }],
+    })
+  }
+
+  if (customerToken) {
+    const customerId =
+      customerToken && (await getCustomerId({ customerToken, config }))
+
+    if (!customerId) {
+      return res.status(400).json({
+        data: null,
+        errors: [{ message: 'Invalid request' }],
+      })
+    }
+
+    let wishlistItems: WishlistItem[] = []
+    let metafieldInput: MetafieldInput = {}
+
+    const { wishlist } = await commerce.getCustomerWishlist({
+      variables: { customerId },
+      config,
+    })
+
+    if (wishlist) {
+      wishlistItems = wishlist?.items?.map(({ product, ...rest }) => {
+        return rest
+      })!
+      metafieldInput = { id: wishlist?.id! }
+    } else {
+      metafieldInput = { namespace: 'my_fields', key: 'wishlist' }
```

Comment:
@chemmanouilidis just a heads up, I believe you need a `type: 'json'` on this `metafieldInput`, otherwise the initial metafield API call will quietly fail and the Wishlist is never created.